client/src/components/game/gui/GUI.jsx

```diff
@@ -49,6 +51,7 @@ function GUI() {
     const [trackedTask, setTrackedTask] = useState(null);
     const [loggedIn, setLoggedIn] = useState(ApplicationState.loggedIn);
     const [targetDungeonPortal, setTargetDungeonPortal] = useState(null);
+    const [inDungeon, setInDungeon] = useState(null);
```

Comment:
I think this should use a boolean for the default value instead of `null`.
While `null` is falsy and does work, when it is used it implies that the value can be an object or some other kind of complex value.
Using `false` here instead would be more explicit that this is a yes/no kind of value.